cmd/common.go

```diff
@@ -51,70 +55,71 @@ func newIAMClient(region string) iamiface.IAMAPI {
 	return iam.New(sess)
 }
 
-func newELBv2Client(region string, cacheCfg *cache.Config) elbv2iface.ELBV2API {
+func newAWSSession(region string) (*session.Session, error) {
 	config := aws.NewConfig().WithRegion(region)
 	config = config.WithCredentialsChainVerboseErrors(true)
+
+	if refreshExpiredCredentials {
+		filename := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+		if filename == "" {
+			filename = defaults.SharedCredentialsFilename()
+		}
+
+		profile := os.Getenv("AWS_PROFILE")
+		if profile == "" {
+			profile = "default"
+		}
+
+		// When corehandlers.AfterRetryHandler calls Config.Credentials.Expire,
+		// the SharedCredentialsProvider forces refreshing credentials from file.
+		// With the SDK's default credential chain, the file is never read.
+		config.WithCredentials(credentials.NewCredentials(&credentials.SharedCredentialsProvider{
+			Filename: filename,
+			Profile:  profile,
+		}))
+	}
+
 	config = request.WithRetryer(config, log.NewRetryLogger(DefaultRetryer))
 	sess, err := session.NewSession(config)
 	if err != nil {
-		log.Fatalf("failed to create elbv2 client, %v", err)
+		return nil, err
+	}
+
+	return sess, nil
+}
+
+func newELBv2Client(region string, cacheCfg *cache.Config) elbv2iface.ELBV2API {
+	sess, err := newAWSSession(region)
+	if err != nil {
+		log.Fatalf("failed to create AWS session, %v", err)
 	}
-	cache.AddCaching(sess, cacheCfg)
```

Comment:
Why are we removing caching here?
caching is required for lifecycle-manager for handling bursts of terminations without being throttled by AWS.
Can you re-add it?